dm/relay/relay.go

```diff
@@ -343,6 +347,18 @@ func (r *Relay) process(ctx context.Context) error {
 						err = err2
 						goto checkError
 					}
+					// only skip events of same GTID after recovering
+					if res.Event.Header.EventType == replication.GTID_EVENT ||
+						res.Event.Header.EventType == replication.MARIADB_GTID_EVENT {
+						gtidStr, err3 := event.GetGTIDStr(res.Event)
+						if err3 != nil {
+							return err3
+						}
+
+						if gtidStr != r.currentGTID {
+							return errors.New("after recover GTID-based replication, the first GTID is not same as broken one")
+						}
+					}
```

Comment:
I don't think we need these extra logic(L334-L361) for duplicate events since we have deal with it in https://github.com/pingcap/ticdc/blob/0076d097b2ab252ca81b7a05430c90f34b73754d/dm/relay/relay_writer.go#L247